File: src/main/java/com/meetify/server/repository/UserRepository.java
package com.meetify.server.repository;

import com.meetify.server.model.Id;
import com.meetify.server.model.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

/**
 * com.meetify.server.repository
 * Created by kr3v on 04.10.2016.
 */

public interface UserRepository extends JpaRepository<User, Long> {
    Optional<User> findById(Id id);
}

File: src/main/java/com/meetify/server/repository/UserRepository.kt
package com.meetify.server.repository

import com.meetify.server.model.User

/**
 * Interface, that used to provide layer between database and server logic by autowiring by server.
 * This interface represents users.
 * @author      Dmitry Baynak
 * @version     0.0.1
 * @since       0.0.1
 */
interface UserRepository : BaseRepository<User>
